build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/test/MutedTestsBuildService.java

```diff
@@ -94,6 +94,16 @@ private List<String> buildExcludePatterns(File file) {
         return excludes;
     }
 
+    private void addExcludes(List<String> excludes) {
```

Comment:
We can eliminate this check by just making `excludePatterns` a `Set` instead. Generally a `HashSet` if we don't care about ordering or a `LinkedHashSet` if we want to preserve insertion order (essentially make it behave like an `ArrayList`).